src/drivers/wayland/lv_wayland.h

```diff
@@ -0,0 +1,153 @@
+ **********************/
+
+/**
+ * @brief Initializes the wayland client
+ * @note must be called after lv_init
+ */
+void lv_wayland_init(void);
+
+/**
+ * @brief De-Initializes the wayland client
+ * @note Must be called on application exit
+ */
+void lv_wayland_deinit(void);
+
+/**
+ * @brief Retrieves the file descriptor of the wayland socket
+ */
+int lv_wayland_get_fd(void);
+
+/**
+ * @brief Creates a window
+ * @param hor_res The width of the window in pixels
+ * @param ver_res The height of the window in pixels
+ * @param title The title of the window
+ * @param close_cb The callback that will be execute when the user closes the window
+ */
+lv_display_t * lv_wayland_window_create(uint32_t hor_res, uint32_t ver_res, char * title,
+                                        lv_wayland_display_close_f_t close_cb);
+
+/**
+ * @brief Closes the window programmatically
+ * @param disp Reference to the LVGL display associated to the window
+ */
+void lv_wayland_window_close(lv_display_t * disp);
+
+/**
+ * @brief Closes the window programmatically
+ * @param disp Reference to the LVGL display associated to the window
+ * @return true: The window is open
+ */
+bool lv_wayland_window_is_open(lv_display_t * disp);
+
+/**
+ * @brief Sets the fullscreen state of the window
+ * @param disp Reference to the LVGL display associated to the window
+ * @param fullscreen If true the window enters fullscreen
+ */
+void lv_wayland_window_set_fullscreen(lv_display_t * disp, bool fullscreen);
+
+/**
+ * @brief Sets the maximized state of the window
+ * @param disp Reference to the LVGL display associated to the window
+ * @param fullscreen If true the window is maximized
+ */
+void lv_wayland_window_set_maximized(lv_display_t * disp, bool maximize);
+
+/**
+ * @brief Obtains the input device of the mouse pointer
+ * @note It is used to create an input group on application start
+ * @param disp Reference to the LVGL display associated to the window
+ * @return The input device
+ */
+lv_indev_t * lv_wayland_get_pointer(lv_display_t * disp);
+
+/**
+ * @brief Obtains the input device of the encoder
+ * @note It is used to create an input group on application start
+ * @param disp Reference to the LVGL display associated to the window
+ * @return The input device
+ */
+lv_indev_t * lv_wayland_get_pointeraxis(lv_display_t * disp);
+
+/**
+ * @brief Obtains the input device of the keyboard
+ * @note It is used to create an input group on application start
+ * @param disp Reference to the LVGL display associated to the window
+ * @return The input device
+ */
+lv_indev_t * lv_wayland_get_keyboard(lv_display_t * disp);
+
+/**
+ * @brief Obtains the input device of the touch screen
+ * @note It is used to create an input group on application start
+ * @param disp Reference to the LVGL display associated to the window
+ * @return The input device
+ */
+lv_indev_t * lv_wayland_get_touchscreen(lv_display_t * disp);
+
+/**
+ * @brief Wrapper around lv_timer_handler
```

Comment:
Please remove the `@brief` tags for consistency with other docs
```suggestion
 * Wrapper around lv_timer_handler
```

Reply:
Ok will remove the @brief, for the stubs since we added extra dependencies to ``install-prerequistes.sh`` I had an idea to generate those files anyways, they are quite large and might change and break the CI several times per year it will probably annoy everyone. So I've added the generation step to ``tests/main.py`` it generates the files and adds guards to them using the ``sed`` utility. (On the weekend, I prefer to avoid Cmake :) ) It's a weekend after all.

Reply:
https://github.com/lvgl/lvgl/pull/6549/commits/bd39614dbdb6617e7efa44ffbbfac6f3b05b386f

Reply:
Nice! It's great that the test uses `wayland-scanner`.